web/packages/core/src/load-ruffle.ts

```diff
@@ -55,8 +61,40 @@ async function fetchRuffle(config: Config): Promise<typeof Ruffle> {
     const { default: init, Ruffle } = await (extensionsSupported
         ? import("../pkg/ruffle_web-wasm_extensions")
         : import("../pkg/ruffle_web"));
+    let response;
+    const wasmUrl = extensionsSupported
+        ? new URL("../pkg/ruffle_web-wasm_extensions_bg.wasm", import.meta.url)
+        : new URL("../pkg/ruffle_web_bg.wasm", import.meta.url);
+    const wasmResponse = await fetch(wasmUrl.href);
+    if (progressCallback) {
+        const contentLength = wasmResponse.headers.get("content-length")!;
+        let bytesLoaded = 0;
+        const bytesTotal = parseInt(contentLength, 10);
+        response = new Response(
+            new ReadableStream({
+                async start(controller) {
+                    const reader = wasmResponse.body?.getReader();
+                    if (!reader) {
+                        throw "Response had no body";
+                    }
+                    progressCallback?.(bytesLoaded, bytesTotal);
```

Comment:
nit: I assme there's no need for the `?.` here (and below) because this code is guarded by `if (progressCallback)`. I'm surprised that TypeScript doesn't warn about this.

Reply:
Results in:

`src/load-ruffle.ts:80:38 - error TS1003: Identifier expected.`

`80                     progressCallback.(bytesLoaded, bytesTotal);`